From a3e4e37ee67be28edc3daa78c97ce9a1e457b0b8 Mon Sep 17 00:00:00 2001
From: Dennis Kempin <denniskempin@chromium.org>
Date: Thu, 7 Jun 2012 17:50:19 -0700
Subject: [PATCH 1/2] Fixed bug: absinfo not initialized

When reading device info from a file, the absinfo structure is not initialized.
This causes fields that are not read from file to be set to undefined values.
Fixed by initializing with zero.
---
 src/evemu.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/evemu.c b/src/evemu.c
index 9d34101..8501933 100644
--- a/src/evemu.c
+++ b/src/evemu.c
@@ -286,7 +286,7 @@ static void read_mask(struct evemu_device *dev, FILE *fp)
 
 static void read_abs(struct evemu_device *dev, FILE *fp)
 {
-	struct input_absinfo abs;
+	struct input_absinfo abs = {0};
 	int index;
 	while (fscanf(fp, "A: %02x %d %d %d %d\n", &index,
 		      &abs.minimum, &abs.maximum, &abs.fuzz, &abs.flat) > 0)
-- 
1.7.7.3

